### PR TITLE
[DowngradePhp80] Handle Attribute in DowngradePropertyPromotionRector

### PR DESCRIPTION
**Pull request by samsonasik**
Fixes #376 Closes https://github.com/rectorphp/rector/issues/6546

### CODE REVIEW

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba it is ready for review.

**Comment by samsonasik** [comment]
Summary:

- Attributes marked as comment
- <strike>- normal comment on property promotion be deleted due to conflict if both attribute and comment exists during marking as comment</strike>
- normal comment on property promotion be moved to property assign in class method stmts
- <strike>both normal comment + Attribute on property promotion exists will be kept and moved to assign as well due marked as comment</strike> 
- for both attribute and comment, the comment will be moved to property assign
- normal comment on normal parameter be kept.

**Comment by samsonasik** [comment]
I refactored to make normal comment on property promotion be moved to property assign in class method stmts 🎉  https://github.com/rectorphp/rector-src/pull/410/commits/9eccb929e06a60fc4e8dd48f9c0c5d7e62c7a427

**Comment by samsonasik** [comment]
both normal comment + Attribute on property promotion exists will be kept and moved to assign as well due marked as comment

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba it is ready for review.

**Comment by samsonasik** [comment]
Finally 🎉🎉🎉, for both attribute and comment, the comment will be moved to property assign

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba it is ready for review.

File: rules-tests/DowngradePhp80/Rector/Class_/DowngradePropertyPromotionRector/Fixture/with_attributes.php.inc [hunk 1]
```diff
@@ -0,0 +1,50 @@
+<?php
+
+namespace Rector\Tests\DowngradePhp80\Rector\Class_\DowngradePropertyPromotionRector\Fixture;
+
+class WithAttributes
+{
+    public function __construct(
+        #[\Deprecated]
+        private int $property = 0,
```
Comment (TomasVotruba) [inline rules-tests/DowngradePhp80/Rector/Class_/DowngradePropertyPromotionRector/Fixture/with_attributes.php.inc:9]: What about case like this?

```php
#[\Deprecated] private int $property = 0,
```

Comment (samsonasik) [reply rules-tests/DowngradePhp80/Rector/Class_/DowngradePropertyPromotionRector/Fixture/with_attributes.php.inc:9]: That will be processed as well https://github.com/rectorphp/rector-src/pull/410/commits/bf8008da8f6124b04add4d67daf4f08c9ebb1413

**Comment by samsonasik** [comment]
Test error notice on `sabbelasichon/typo3-rector` seems unrelated as I only update test fixture after last green, @sabbelasichon could you verify? Thank you.

All other checks have passed 🎉 @TomasVotruba I think it is ready.

**Comment by TomasVotruba** [comment]
> Test error notice on sabbelasichon/typo3-rector seems unrelated as I only update test fixture after last green, @sabbelasichon could you verify? Thank you.

I think that might be releated to new php-parser version, released 6 days ago https://github.com/nikic/PHP-Parser/releases/tag/v4.11.0


Merging, thank you :heavy_check_mark:

**Pull request merged**